concept/type/impl/TypeImpl.java

```diff
@@ -149,6 +150,7 @@ <TYPE extends TypeImpl> Forwardable<TYPE, Order.Asc> getSubtypesExplicit(Functio
     }
 
     void validateDelete() {
+        if (isRoot()) throw exception(TypeDBException.of(ROOT_TYPE_MUTATION));
```

Comment:
Not strictly necessary as long as all `Root` types are covered, but I do think that having this unconditional validation step is nice to have.